Godeps/_workspace/src/github.com/docker/distribution/registry/handlers/layerupload.go

```diff
@@ -171,6 +171,9 @@ func (luh *layerUploadHandler) PatchLayerData(w http.ResponseWriter, r *http.Req
 		return
 	}
 
+	// Flush the upload layer data before sending a reponse
```

Comment:
Technically, because this is deferred, the response will be sent before the flush happens.